### PR TITLE
Update the destination repository during the sync command and add description metadata

### PR DESCRIPTION
**Pull request by jycamier**
As a GHES user, I need to get descriptions repositories from public actions.
As a GHES user, I want to update these descriptions.

### CODE REVIEW

**Review by ajaykn — COMMENTED** [review]
Thanks for the valuable contribution.

Here are few things we would like to know more on the problem statement.
> As a GHES user, I need to get descriptions repositories from public actions.

Do you have a requirement to use these descriptions of the public actions somewhere in your enterprise workflows.

Apart from that, 
- Please update the PR title to more descriptive.

File: src/push.go [hunk 1]
```diff
@@ -86,7 +86,18 @@ func PushManyWithGitImpl(ctx context.Context, flags *PushFlags, repoNames []stri
 }
 
 func PushWithGitImpl(ctx context.Context, flags *PushFlags, repoName string, ghClient *github.Client, gitimpl GitImplementation) error {
-	_, nwo, err := extractSourceDest(repoName)
+	org, nwo, err := extractSourceDest(repoName)
+	if err != nil {
+		return err
+	}
+
+	orgOwnerName, orgRepoName, err := splitNwo(org)
```
Comment (ajaykn) [inline src/push.go:94]:
```suggestion
	sourceOwnerName, sourceRepoName, err := splitNwo(source)
```

File: src/push.go [hunk 2]
```diff
@@ -86,7 +86,18 @@ func PushManyWithGitImpl(ctx context.Context, flags *PushFlags, repoNames []stri
 }
 
 func PushWithGitImpl(ctx context.Context, flags *PushFlags, repoName string, ghClient *github.Client, gitimpl GitImplementation) error {
-	_, nwo, err := extractSourceDest(repoName)
+	org, nwo, err := extractSourceDest(repoName)
```
Comment (ajaykn) [inline src/push.go:89]:
```suggestion
	source, destination, err := extractSourceDest(repoName)
```
this would be more clear

Comment (jycamier) [reply src/push.go:89]: Sure but if we change `nwo` to `destination`, there are a bunch of changes to add in the code, not related to my PR and we break the global logical naming rules in the project. Are you sure about it ?

File: src/push.go (hunk 1 above)
Comment (jycamier) [reply src/push.go:94]: Sure but if we change `org` to `source`, there are a bunch of changes to add in the code, not related to my PR and we break the global logical naming rules in the project. Are you sure about it ?

Here's an example of existing code in the same file which use `org` to refer about

```go
// check if the owner refers to the authenticated user or an organization.
var createRepoOrgName string
if strings.EqualFold(*currentUser.Login, ownerName) {
    // we'll create the repo under the authenticated user's account.
    createRepoOrgName = ""
} else {
    // ensure the org exists.
    createRepoOrgName = ownerName
    _, err := getOrCreateGitHubOrg(ctx, client, ownerName, *currentUser.Login)
    if err != nil {
	return nil, err
    }
}
```

**Comment by jycamier** [comment]
> Do you have a requirement to use these descriptions of the public actions somewhere in your enterprise workflows.

Sure. When you synchronize a set of public actions in a GHES instance, the name is not sufficient to know what the action is doing. It is very useful for technicians to check if there is an existing action that is already doing the job, instead of asking to sync another action that does the same.

In my workflow, I create an auto-generated documentation about all actions available in our GHES.

File: src/push.go (hunk 1 above)
Comment (jycamier) [reply src/push.go]: Done !

**Comment by jycamier** [comment]
@ajaykn I've change some naming.

**Comment by ajaykn** [comment]
> @ajaykn I've change some naming.

Looks good now thanks :)

**Comment by ajaykn** [comment]
@jycamier Thanks for addressing comments, currently ci build is failing, i am looking into that, once resolved, will merge and release :)

**Comment by jycamier** [comment]
@ajaykn indeed.
On my fork which we currently use, I have add this environment variable

https://github.com/jycamier/actions-sync/commit/69b813c7fac5aee2ae70dd220689255e01a07683

**Comment by ajaykn** [comment]
I was working on CI build and its was failing due to old go version and fixed here: https://github.com/actions/actions-sync/pull/30
Since the CI build has succeeded in main branch, Can you please pull the latest main and update ?

**Comment by jycamier** [comment]
@ajaykn sorry about the delay
The rebase from your code is done.

**Comment by ajaykn** [comment]
Interesting, CI build is still failing. Will have a look and update.
Thanks for patience.

**Comment by ajaykn** [comment]
@jycamier Apologies, its been a while. Hope you are doing well too.
I have checked CI build and its failing due to the changes introduced in this PR.

You would be facing the same when u run locally `docker-compose run --rm test-build`

As part of the push tests, we are seeing the below error
https://github.com/actions/actions-sync/pull/28/checks#step:5:29
```
FAIL: Failed pushing new commit to main
-----output-----
GET https://api.github.com/repos/org/repo: 404 Not Found []
-----/output-----
-----github git daemon output-----
```

Here is the test failing, https://github.com/actions/actions-sync/blob/main/script/test-build#L43-L44

With the changes in PR, you are trying to fetch the repo details from github.com api, and since in the test case, the repo_name is `org/repo` which doesnt exists, so it's throwing NOT FOUND.

Also I see a major concern with the changes after thorough review and confirmation.
As per the docs quote here in banner:
https://docs.github.com/en/enterprise-server@3.2/admin/github-actions/managing-access-to-actions-from-githubcom/manually-syncing-actions-from-githubcom#example-using-the-actions-sync-tool


> Note: This example uses the actions-sync sync command, which requires concurrent access to both the GitHub.com API and your enterprise instance's API from your machine. If you can only access one system at a time, you can use the actions-sync pull and push commands.

So customers use pull and push separately when they can access only one system at a time from their machine,  and in that case, we wont even access github.com apis as part of PUSH and current PR would break those changes.


If you still see the value in adding description while syncing, may be you can do the below solution
- Apply try catch and if GET on github.com api fails, then don't break the push flow but continue without the desciption (which is existing behavior)

Please let me know if you think of any alternate solution.

**Review by ajaykn — CHANGES_REQUESTED** [review]
Since its a breaking change for `push`, please check  https://github.com/actions/actions-sync/pull/28#issuecomment-1073073552 for more details

**Comment by jycamier** [comment]
@ajaykn sorry 
I haven't seen your message ! I'll check it

**Comment by shawnHartsell** [comment]
@jycamier thank you for the contribution! Unfortunately the ownership of this repo has changed hands several times since the opening of this PR. Going forward we will be focusing mainly on maintenance and bug fixes before accepting new features. 

https://github.com/actions/actions-sync/discussions/85

If you still feel like this is a worthwhile pursuit could you create a new feature request issue and link this PR to it? Thanks!

**Pull request closed without merging**